dotnet/src/Connectors/Connectors.GoogleVertexAI/GeminiPromptExecutionSettings.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+
+/// <summary>
+/// Represents the settings for executing a prompt with the Gemini model.
+/// </summary>
+[JsonNumberHandling(JsonNumberHandling.AllowReadingFromString)]
+public sealed class GeminiPromptExecutionSettings : PromptExecutionSettings
+{
+    private double? _temperature;
+    private double? _topP;
+    private int? _topK;
+    private int? _maxTokens;
+    private int? _candidateCount;
+    private IList<string>? _stopSequences;
+    private IList<GeminiSafetySetting>? _safetySettings;
+    private ToolCallBehavior? _toolCallBehavior;
+
+    /// <summary>
+    /// Default max tokens for a text generation.
+    /// </summary>
+    public static int DefaultTextMaxTokens { get; } = 256;
+
+    /// <summary>
+    /// Temperature controls the randomness of the completion.
+    /// The higher the temperature, the more random the completion.
+    /// Range is 0.0 to 1.0.
+    /// </summary>
+    [JsonPropertyName("temperature")]
+    public double? Temperature
+    {
+        get => this._temperature;
```

Comment:
Is there no default value defined for temperature? What happens when this is null?

Reply:
Any of params can be null then gemini takes default for used model. Defaults are described in google docs.